src/Microsoft.Health.Fhir.SqlServer/Features/Schema/SchemaInitializer.cs

```diff
@@ -59,18 +71,18 @@ internal void Initialize(bool forceIncrementalSchemaUpgrade = false)
                 else
                 {
                     // Apply the maximum supported version. This won't consider the .diff.sql files.
-                    _schemaUpgradeRunner.ApplySchema((int)_schemaInformation.MaximumSupportedVersion, applyFullSchemaSnapshot: true);
+                    _schemaUpgradeRunner.ApplySchema(_schemaInformation.MaximumSupportedVersion, applyFullSchemaSnapshot: true);
                 }
 
                 GetCurrentSchemaVersion();
             }
 
             // If the current schema version needs to be upgraded
-            if (_schemaInformation.Current < _schemaInformation.MaximumSupportedVersion)
+            if (_sqlServerDataStoreConfiguration.SchemaUpdatesEnabled && _schemaInformation.Current < _schemaInformation.MaximumSupportedVersion)
             {
                 // Apply each .diff.sql file one by one.
-                int current = (int?)_schemaInformation.Current ?? 0;
-                for (int i = current + 1; i <= (int)_schemaInformation.MaximumSupportedVersion; i++)
+                int current = _schemaInformation.Current ?? 0;
```

Comment:
This part is a bit confusing. Can `_schemaInformation.Current` be null here? If so, then the above comparison in the `if` condition would not have worked right? If it cannot be null, then we can directly use it?